lib/trino-collect/src/main/java/io/trino/collect/cache/NonEvictableCache.java

```diff
@@ -15,12 +15,26 @@
 
 import com.google.common.cache.Cache;
 
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+
 /**
  * A {@link com.google.common.cache.Cache} that does not support eviction.
  */
 public interface NonEvictableCache<K, V>
         extends Cache<K, V>
 {
+    default V getUnchecked(K key, Supplier<V> loader)
+    {
+        try {
+            return get(key, loader::get);
+        }
+        catch (ExecutionException e) {
+            // this can not happen because a supplier can not throw a checked exception
+            throw new RuntimeException("Unexpected checked exception from cache load", e);
```

Comment:
can -> should

(actually "it can happen", if you do some things you shouldn't do)